device/globals/item.gd

```diff
@@ -282,10 +288,22 @@ func _update_terrain():
 	var color = terrain.get_terrain(pos)
 	var scale_range = terrain.get_scale_range(color.b)
 
+	scale_range.x *= pose_scale
 	if scale_on_map && (self is Node2D) && scale_range != get_scale():
-		var c = terrain.get_terrain(pos)
-		var s = terrain.get_scale_range(c.b)
-		set_scale(s)
+		# Scan children to see if `interact_pos` is a child of ours, and if so,
+		# take a backup of the global position, because it will be affected by scaling.
+		var interact_global_position
+		for child in get_children():
+			if child == interact_pos:
+				interact_global_position = interact_pos.get_global_position()
+				break
```

Comment:
I think it's OK to assume that `interact_pos` is a direct child of the item scene, unless we're assuming that it might not be elsewhere. If a developer runs into problems with this, it will be obvious enough that they'll check on the documentation, and I wrote a doc on the typical anatomy of an item where it could be emphasised that it should be a direct child. I'm mentioning it because it would probably save some complexity.